src/installer/managed/Microsoft.NET.HostModel/Bundle/Bundler.cs

```diff
@@ -49,17 +53,67 @@ public class Bundler
             RuntimeConfigJson = appAssemblyName + ".runtimeconfig.json";
             RuntimeConfigDevJson = appAssemblyName + ".runtimeconfig.dev.json";
 
-            BundleManifest = new Manifest(Target.BundleVersion, netcoreapp3CompatMode: options.HasFlag(BundleOptions.BundleAllContent));
+            BundleManifest = new Manifest(Target.BundleMajorVersion, netcoreapp3CompatMode: options.HasFlag(BundleOptions.BundleAllContent));
             Options = Target.DefaultOptions | options;
         }
 
+        private bool ShouldCompress(FileType type)
+        {
+            // compression is not supported before bundle vesion 6
+            if (Target.BundleMajorVersion < 6)
+            {
+                return false;
```

Comment:
I wonder if we should throw if compression is enabled and target version is < 6.
SDK will need to have checks for this anyway, but still. The current behavior is "silently don't do it" - which is always a bit strange.

Reply:
Also this check should be done once up-front and not per-file-type.

Reply:
This was before it was decided that the caller (SDK or tests) needs to pass a flag. I think it can be an assert now - if the flag is on, it is better be 6+

Reply:
> I wonder if we should throw if compression is enabled and target version is < 6

Right, since the arguments are passed in from external user (even that is SDK), it should not be assert. 

`ArgumentException("Compression requires framework version 6.0 or above", nameof(options));`